Misc/NEWS

```diff
@@ -1100,6 +1100,10 @@ Documentation
 Build
 -----
 
+- bpo-29941: Add ``--with-assertions`` configure flag to explicitly enable
+  C asertions. Defaults to off, but ``--with-pydebug`` also enables
```

Comment:
C assert() checks via NDEBUG.  Defaults to off.  --with-pydebug implies --with-assertions.

Side note: are we supposed to put markdown backticks in news entries now? I see others with them so I guess it doesn't hurt. But given I'm used to viewing it as UTF-8 rather than MD it looks suspicious.  I don't care either way.